src/core/blockchain/ethereum/name-service.ts

```diff
@@ -38,27 +46,19 @@ export class NameService implements IBlockchainNameService {
             }
         }
     }
-
-    public async resolveName(name: string): Promise<IResolveNameResponse> {
-        const node = namehash.hash(name);
-        let data = await this.client.http.jsonRpc('eth_call', [
-            {
-                to: ENS,
-                data: '0x' + simpleEncode('resolver(bytes32)', node).toString('hex')
-            },
-            'latest'
-        ]);
-        const resolverAddress =
-            '0x' + rawDecode(['address'], Buffer.from(data.result.replace('0x', ''), 'hex'))[0];
-        data = await this.client.http.jsonRpc('eth_call', [
-            {
-                to: resolverAddress,
-                data: '0x' + simpleEncode('addr(bytes32)', node).toString('hex')
-            },
-            'latest'
-        ]);
-        const address =
-            '0x' + rawDecode(['address'], Buffer.from(data.result.replace('0x', ''), 'hex'))[0];
-        return Promise.resolve({ address });
+    public resolveName(name: string): Promise<IResolveNameResponse> {
+        const { mainNet } = this.client.network;
+        const ending = name.split('.').pop();
+        const { service, record } = config.nameServices.find(item =>
+            typeof item.tld === 'string' ? item.tld === ending : !!ending.match(item.tld)
+        );
+        switch (service) {
```

Comment:
this method is basically the same with zilliqa one, move it in `GenericNameService`

Reply:
This is where the tests fail. I know it's duplicated, but this is what me and George came up with in order to solve it